input-explainer.md

```diff
@@ -280,6 +294,91 @@ When the canvas receives a `pointerdown` event an `XRInputSource` is created wit
+
+Each input also has a `name` which is a human readable label that should correspond to the name the platform gives for the input if possible. For example, inputs in the `extended` array may have names like "A", "B", "X", and "Y". If a platform-specific name cannot be determined the name should reflect the input's semantic type, such as "Touchpad" or "Button 0".
+
+```js
+function onXRFrame(timestamp, frame) {
+  let inputSource = xrSession.getInputSources()[0];
+
+  // Check to see if the input source is a controller.
+  if (inputSource.controller) {
+    
+    // Determine if the controller has a joystick with at least 2 axes.
+    let joystick = inputSource.controller.joystick;
+    if (joystick && joystick.value.length >= 2) {
+      MoveUser(joystick.value[0], joystick.value[1]);
+    }
+
+    // Determine if the controller has at least one extended input, such as a
+    // face button.
+    if (inputSource.controller.extended.length > 0) {
+      let extended0 = inputSource.controller.extended[0];
+      if (extended0.pressed) {
+        EmitPaint();
+      }
+    }
+    
+    // etc.
+  }
+
+  // Do the rest of typical frame processing...
+}
+```
+
+If the application includes interactions that require user activation (such as starting media playback), the application can listen to the `XRInputSource`s `select` events, which fire for every pressed and released input on the controller. When triggered by a controller input, the `XRInputSourceEvent` will include a non-null `XRTrackedControllerState` to indicate which input triggered the event.
+
+The UA may update the controller state at any point, which the restriction that it must remain constant during the running of each batch of `XRSession` `requestAnimationFrame` callbacks.
+
+> Note: You might wonder why `XRTrackedController` isn't tied to an `XRFrame` like the `XRInputPose` is. The answer is that data exposed by `XRFrame` is intended to be highly timing dependent, and have a limited useful lifespan. Poses are continuously changing, demand low latency, and are typically projected forward to a specific point in time. The also must be queried relative to a specific `XRFrameOfReference`. By contrast, the values reported by a `XRTrackedControllerState` change with comparatively lower frequency, are less sensitive to latency requirements, and can be reasonably inspected outside the core frame loop without concerns of the data becoming unusably stale, and carry no spatial data so there's no need to associate them with an `XRFrameOfReference`. As such, it's preferable to aim fot the better API ergonomics offered by making `XRTrackedController` more persistently available. 
+>
```

Comment:
"must remain constant during Javascript execution" is basically the default assumption of Javascript, isn't it? Being single threaded, it would be highly unusual if values did change mid-execution without an obvious call to update them. rAF is special simply in that during the processing of rAF callbacks control is returned to the browser multiple times, once per callback, and yet we want the controller state to be consistent across all of them.

I'm looking at an alternative way of approaching this functionality, but I'll try to update all of this language to be clearer and take into account the rest of your feedback if it carries through to the next update.